core/src/main/scala/cats/syntax/foldable.scala

```diff
@@ -47,7 +47,8 @@ private[syntax] trait FoldableSyntaxBinCompat1 {
 }
 
 final class NestedFoldableOps[F[_], G[_], A](private val fga: F[G[A]]) extends AnyVal {
-  def sequence_(implicit F: Foldable[F], G: Applicative[G]): G[Unit] = F.sequence_(fga)
+  def sequence_(implicit F: Foldable[F], G: Applicative[G], ev: A <:< Unit): G[Unit] =
+    F.sequence_(fga.asInstanceOf[F[G[Unit]]])
```

Comment:
Right, we cannot change the signature of this method. What we can do is add a new ops class:

```scala
final class NestedUnitFoldableOps[F[_], G[_]](private val fgu: F[G[Unit]])
```

Reply:
so this one has to change then to `F.sequence_(G.void(fga))`?

Reply:
Seems so.

Reply:
That still allows people to call `.sequence` on non-unit structures -- I would suggest deprecating it

Reply:
We should deprecate it, and remove `implicit` from the arguments. This will break existing uses.

Reply:
though even that seems so knotty I'm not sure it's worthwhile.

@armanbilge what do you think of instead of this approach, adding new methods (and maybe deprecating the old ones):

```
def traverseEach[G[_]: Monad](fga: F[G[A]])(f: A => G[Unit]): G[Unit] = foldMapM(fga)(f)
def sequenceEach[G[_]: Monad](fgu: F[G[Unit]): G[Unit].= traverseEach(fgu)(identity)
def traverseEachA[G[_]: Applicative](fga: F[G[A]])(f: A => G[Unit]): G[Unit] = foldMapA(fga)(f) // potentially less performant, prefer traverseEach if monad is available
def sequenceEachA[G[_]: Applicative](fgu: F[G[Unit]]): G[Unit] = traverseEachA(fga)(identity)
```

Reply:
> though even that seems so knotty I'm not sure it's worthwhile.

honestly, this is my feeling about this PR as a whole. It's obvious at the moment we don't even have existing test coverage to safely make this change.

Reply:
if we can't change signatures, what about adding new methods with an appropriate signature and deprecating the old ones?

Reply:
@armanbilge 
> It's obvious at the moment we don't even have existing test coverage to safely make this change.

Do you mean, there's no tests for `traverse_` nor `sequence_` or do you mean some specific test scenarios?

Reply:
@satorg see https://github.com/typelevel/cats/pull/4352#discussion_r1027255402